modules/apps/jaggery/mdm/src/mdm/modules/backend-service-invoker.js

```diff
@@ -56,20 +57,22 @@ var backendServiceInvoker = function () {
                 } else {
                     return successCallback(null);
                 }
-            } else if (xmlHttpRequest.status == 401) {
+            } else if (xmlHttpRequest.status == 401 && (xmlHttpRequest.responseText == TOKEN_EXPIRED ||
+                                                        xmlHttpRequest.responseText == TOKEN_INVALID )) {
                 tokenUtil.refreshToken();
-                return execute();
-            }
-            else {
+                return execute(count);
+            } else {
                 return errorCallback(parse(xmlHttpRequest.responseText));
             }
         };
         var accessToken = session.get(constants.ACCESS_TOKEN_PAIR_IDENTIFIER).accessToken.trim();
-        if (accessToken){
-            return execute();
-        }else {
-            tokenUtil.refreshToken();
-            return execute();
+        if (accessToken) {
+            return execute(1);
+        } else {
+            var dummyRespose = {};
+            dummyRespose.status = 401;
```

Comment:
Noted.
The dummy Response was removed after the IS version bump.
Current "backend-service-invoker.js" doesn't have the above code snippet.